src/application-base/private/useTestOverrides.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+
+const APPLICATION_BASE_OVERRIDE_EVENT = 'applicationBase.testOverride';
+
+/**
+ * Custom React hook that overrides locale via events. Leveraged by full stack testing to control deployed applications.
+ * @returns {Object} Contains the locale to override with.
+ */
+const useTestOverrides = () => {
+  const [localeOverride, setLocaleOverride] = useState();
+
+  const handleTestOverrides = (event) => {
+    if (!event || !event.metaData) {
+      return;
+    }
+
+    const { metaData } = event;
+    if (typeof metaData.locale !== 'undefined') {
+      setLocaleOverride(metaData.locale);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener(APPLICATION_BASE_OVERRIDE_EVENT, handleTestOverrides);
+    return () => {
+      window.removeEventListener(APPLICATION_BASE_OVERRIDE_EVENT, handleTestOverrides);
+    };
+  }, []); // Empty array ensures we setup and teardown event listener once.
```

Comment:
We might need our resident hooks expert to double check the function references won't be an issue here. The `handleTestOverrides` will be redefined on each render, but the useEffect listener is only initialized once. 

cc: @tbiethman

Reply:
It'd work fine, as the correct function instance would be captured in the arrow function at the time it is created. But as it is written, we WOULD create a new function every time the hook is executed and just do nothing with it. Not going to be a performance bottleneck or anything, but it's kind of confusing to look at.